Add test and deploy workflows

p-galligan opened the pull request:
Adds workflows for tests and deploys. 

Removes travis file.

p-galligan committed:
lambda workflow build test

Tests workflow for installing python and running tox

p-galligan committed:
Forgot to add trigger

Adds trigger to action on PR to dev

p-galligan committed:
add direct cache

Add cache logic to get around search for requirements.txt

p-galligan committed:
clone repo and make zip

Clone deploy scripts and make a zip in workflows

p-galligan committed:
Set env variables correctly

Fix environment variables in workflows.

p-galligan committed:
trying again

trying variable assignment again...

p-galligan committed:
test environment variables

Testing env variables

p-galligan committed:
again

trying again

p-galligan committed:
enable zip action

Enables zip and envsubst actions in workflow. Resets basename variable

p-galligan committed:
test working directory

prints working directory in workflow

p-galligan committed:
print ls

Gives an LS of our current directory

p-galligan committed:
testing

test

p-galligan committed:
tweak handler code

Tweak environment variable

p-galligan committed:
checkout code

I forgot to checkout the repo for deploy

p-galligan committed:
Enable push to S3

Enables push to S3 workflow

p-galligan committed:
fix environment

Fix setting environment in workflow

p-galligan committed:
failure test

removing environment setting to test

p-galligan committed:
Enable environment

Re-enable environment

p-galligan committed:
Re-arrange steps

Clone deploy scripts first.

p-galligan committed:
change aws configure

Changes variable assignment for configuring AWS

p-galligan committed:
Enable s3 push

Enables push to S3 in workflows

p-galligan committed:
remove unnecessary line

Removes an unnecessary line from workflow

p-galligan committed:
Enable lambda deploy

Enables lambda deploy in workflows

p-galligan committed:
Remove unnecessary code

Removes unnecessary steps.

p-galligan committed:
Splits workflows

Splits workflows and removes travis file

p-galligan committed:
Merge branch 'development' into workflows

p-galligan committed:
remove dependency from deploy

Removes the dependency from the deploy action and removes travis file

p-galligan committed:
tests deploy

testing deploy workflow

p-galligan committed:
adds conditional back

Adds conditionals back into deploy action

p-galligan committed:
fix typo

fixes typo in the environment of deploy

p-galligan committed:
update deploy workflow

Removes deploy on PR to dev

helrond commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,49 @@
+name: deploy
```
```suggestion
name: Deploy
```

helrond commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,49 @@
+
+jobs:
+  deploy:
+    if: github.ref == 'refs/heads/base' || github.ref == 'refs/heads/development'
```
Is this necessary?

helrond commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,35 @@
+
+jobs:
+  build:
+    if: github.ref != 'refs/heads/base' || github.ref != 'refs/heads/development'
```
Is this necessary?

helrond commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,35 @@
+    runs-on: ubuntu-latest
+
+    environment:
+      name: development
```
Again, wondering if we should use a variable here just in case?

helrond commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,35 @@
+name: build_test
```
```suggestion
name: Tests
```

helrond reviewed (changes requested):
Some really minor things.

p-galligan committed:
update workflows

Removes conditionals from jobs and tweaks the names.

Adds a repository variable for test environment.

p-galligan committed:
Update deploy workflow

Updates the deploy workflow to use role assumption
